File: src/tools/registry.ts
import type { Tool, ToolDefinition } from './base.js';

export class ToolRegistry {
  private tools: Map<string, Tool> = new Map();
  
  register(tool: Tool): void {
    this.tools.set(tool.definition.name, tool);
  }
  
  get(name: string): Tool | undefined {
    return this.tools.get(name);
  }
  
  list(): ToolDefinition[] {
    return Array.from(this.tools.values()).map(t => t.definition);
  }
  
  has(name: string): boolean {
    return this.tools.has(name);
  }
}

export const globalTools = new ToolRegistry();

File: src/tools/registry.py
from .base import Tool, ToolDefinition


class ToolRegistry:
    def __init__(self):
        self._tools: dict[str, Tool] = {}
    
    def register(self, tool: Tool) -> None:
        self._tools[tool.definition.name] = tool
    
    def get(self, name: str) -> Tool | None:
        return self._tools.get(name)
    
    def list(self) -> list[ToolDefinition]:
        return [t.definition for t in self._tools.values()]
    
    def has(self, name: str) -> bool:
        return name in self._tools


global_tools = ToolRegistry()
